backend/src/bootstrap/neode.js

```diff
@@ -0,0 +1,88 @@
+import Neode from 'neode'
+import uuid from 'uuid/v4'
+
+export default function setupNeode(options) {
+  const { uri, username, password } = options
+  const neodeInstance = new Neode(uri, username, password)
+  neodeInstance.model('InvitationCode', {
+    createdAt: { type: 'string', isoDate: true, default: () => new Date().toISOString() },
+    token: { type: 'string', primary: true, token: true },
+    generatedBy: {
+      type: 'relationship',
+      relationship: 'GENERATED',
+      target: 'User',
+      direction: 'in',
+    },
+    activated: {
+      type: 'relationship',
+      relationship: 'ACTIVATED',
+      target: 'EmailAddress',
+      direction: 'out',
+    },
+  })
+  neodeInstance.model('EmailAddress', {
+    email: { type: 'string', primary: true, lowercase: true, email: true },
+    createdAt: { type: 'string', isoDate: true, default: () => new Date().toISOString() },
+    verifiedAt: { type: 'string', isoDate: true },
+    nonce: { type: 'string', token: true },
+    belongsTo: {
+      type: 'relationship',
+      relationship: 'BELONGS_TO',
+      target: 'User',
+      direction: 'out',
+    },
+  })
+  neodeInstance.model('User', {
+    id: { type: 'string', primary: true, default: uuid }, // TODO: should be type: 'uuid' but simplified for our tests
+    actorId: { type: 'string', allow: [null] },
+    name: { type: 'string', min: 3 },
```

Comment:
> [<img alt="Tirokk" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/Tirokk) **Authored by [Tirokk](https://github.com/Tirokk)**
_<time datetime="2019-07-04T06:39:33Z" title="Thursday, July 4th 2019, 8:39:33 am +02:00">Jul 4, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -0,0 +1,88 @@
+import Neode from 'neode'
+import uuid from 'uuid/v4'
+
+export default function setupNeode(options) {
+  const { uri, username, password } = options
+  const neodeInstance = new Neode(uri, username, password)
+  neodeInstance.model('InvitationCode', {
+    createdAt: { type: 'string', isoDate: true, default: () => new Date().toISOString() },
+    token: { type: 'string', primary: true, token: true },
+    generatedBy: {
+      type: 'relationship',
+      relationship: 'GENERATED',
+      target: 'User',
+      direction: 'in',
+    },
+    activated: {
+      type: 'relationship',
+      relationship: 'ACTIVATED',
+      target: 'EmailAddress',
+      direction: 'out',
+    },
+  })
+  neodeInstance.model('EmailAddress', {
+    email: { type: 'string', primary: true, lowercase: true, email: true },
+    createdAt: { type: 'string', isoDate: true, default: () => new Date().toISOString() },
+    verifiedAt: { type: 'string', isoDate: true },
+    nonce: { type: 'string', token: true },
+    belongsTo: {
+      type: 'relationship',
+      relationship: 'BELONGS_TO',
+      target: 'User',
+      direction: 'out',
+    },
+  })
+  neodeInstance.model('User', {
+    id: { type: 'string', primary: true, default: uuid }, // TODO: should be type: 'uuid' but simplified for our tests
+    actorId: { type: 'string', allow: [null] },
+    name: { type: 'string', min: 3 },
```


Ah, I see. Here you constraint the user name length …